### ISSUE TITLE
whenNotEmpty() NOT available after validateIfString()

### ISSUE THREAD

**Issue by jezzsantos** [markusbohl/fluent-ts-validator#4]
(latest Version 2.1.0)

I have this dto:
```
export class ClassA{
 name: string = ""; //<- notice the default value is empty string
}
```

and now its validator:
```
export class ClassAValidator extends AbstractValidator<ClassA> {
    constructor() {
        super();

        this.validateIfString(dto => dto.name)
            .hasLengthBetween(1, 200)
            .whenNotEmpty() //<- not available here
            .withFailureMessage(`name must be 1 to 200 characters`);
    }
```
I only want to validate the `name` if the value is: **NOT undefined AND NOT null AND NOT empty**

But the call to `whenNotEmpty()` is no longer there, as per examples in [documentation](https://github.com/markusbohl/fluent-ts-validator#conditional-validation).

**Comment by markusbohl** [comment]
Thank you for bringing it up. I'll have a look at this issue.

**Comment by markusbohl** [comment]
Issue is fixed with version [2.1.1](https://github.com/markusbohl/fluent-ts-validator/releases/tag/2.1.1). The same problem did not only occur after `validateIfString()` but also after `validateIfIterable()`.
